File: dash_docs/chapters/basic_callbacks/examples/basic-input.R
library(dash)
library(dashCoreComponents)
library(dashHtmlComponents)

app <- Dash$new()

app$layout(
  htmlDiv(
    list(
      dccInput(id = 'input-1', type = 'text', value = 'Montreal'),
      dccInput(id = 'input-2', type = 'text', value = 'Canada'),
      htmlDiv(id = 'output_keywords')
    )
  )
)


app$callback(output(id = 'output_keywords', property = 'children'),
             list(input(id = 'input-1', property = 'value'),
                  input(id = 'input-2', property = 'value')),
             function(input1, input2) {
               sprintf("Input 1 is \"%s\" and Input 2 is \"%s\"", input1, input2)
             })

app$run_server()

File: dash_docs/chapters/basic_callbacks/examples/basic-input.jl
using Dash, DashHtmlComponents, DashCoreComponents


app = dash()

app.layout = html_div() do
    dcc_input(id = "input-10", type = "text", value = "Montreal"),
    dcc_input(id = "input-20", type = "text", value = "Canada"),
    html_div(id = "output-keywords1")
end

callback!(
    app,
    Output("output-keywords1", "children"),
    Input("input-10", "value"),
    Input("input-20", "value"),
) do input_1, input_2
    return "Input 1 is \"$input_1\" and Input 2 is \"$input_2\""
end

run_server(app, "0.0.0.0", debug=true)
